test/test_writer.py

```diff
@@ -198,8 +198,8 @@ def test_wrapper_bug():
 @tdir
 def test_wrapper_bug2():
     with pytest.raises(NotImplementedError) as e:
-        fp = open(FILENAME, 'w')
-        safer.writer(fp, close_on_exit=True, temp_file=True)
+        with open(FILENAME, 'w') as fp:
```

Comment:
Aha, I'm pretty sure you've found the issue!

---

So this change needs two non-obvious tweaks.

I didn't accidentally forget to `open` `fp` in a context in these - I'm using the `close_on_exit` flag, which should automatically close `fp`.

In this first test, I think you have actually revealed an issue:

`with safer.writer(fp, close_on_exit=True, temp_file=True) as writer:`might not be closing `fp` if there's an exception. I need to fix that. (It doesn't get detected in the Linux tests because the file handle gets closed immediately it gets destructed.)

So I need to make this change, or verify the bug doesn't exist, and then the first piece of code needs to change to:

```
fp = open(FILENAME, 'w')
with safer.writer(fp, close_on_exit=True, temp_file=True):
    pass
```

Probably the second code block should stay the same...

I'll make that change and let you know!

Reply:
Nice! Our bug hunt pays off. :)